common/network-common/src/main/java/org/apache/spark/network/crypto/README.md

```diff
@@ -1,6 +1,20 @@
-Forward Secure Auth Protocol
+Forward Secure Auth Protocol v1.1
 ==============================================
 
+Deprecation Notice
+------------------
+This is a bespoke key exchange protocol that was implemented before Spark supported TLS (aka SSL) for RPC
+calls. It is recommended that Spark users upgrade to using TLS for RPC calls between Spark processes. This protocol
+will be deprecated and removed in the long-term.
```

Comment:
To @sweisdb , may I ask when you think we can delete this? If you think `Spark 5`, please remove this claim. It's too far. In Apache Spark community, deprecation doesn't mean the removal.
> and removed in the long-term.

FYI, here is the guideline.
- https://spark.apache.org/versioning-policy.html
https://spark.apache.org/versioning-policy.html

Reply:
After seeing that we're still supporting legacy SASL, I agree that this will likely not be removed any time in the forseeable future.

Reply:
Fixing this.